src/main/interface/msp.c

```diff
@@ -1899,7 +1899,7 @@ static mspResult_e mspProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
         gpsSol.numSat = sbufReadU8(src);
         gpsSol.llh.lat = sbufReadU32(src);
         gpsSol.llh.lon = sbufReadU32(src);
-        gpsSol.llh.alt = sbufReadU16(src);
+        gpsSol.llh.alt = sbufReadU32(src);
```

Comment:
This will break backwards compatibility of the MSP protocol. The right way to this is to leave the existing `uint16_t` untouched and add the `uint32_t` at the end, like done here for `dtermSetpointWeight`: 

https://github.com/AirBreak69/betaflight/blob/0cabe7e70caa7be21336c5c668b3e4ea7e98fa33/src/main/interface/msp.c#L1743-L1759

Also, `gpsSol.llh.alt` needs to be written as well, the same deal re not breaking backwards compatibility applies there as well:

https://github.com/AirBreak69/betaflight/blob/0cabe7e70caa7be21336c5c668b3e4ea7e98fa33/src/main/interface/msp.c#L1036

Reply:
Is 6500 m really that limiting?
Just changing it to int16_t may do (or split positive/negative differently - 5km positive, 1.5535km negative)

Reply:
@mikeller Please check if I implemented the uint32_t versions correctly.
@ledvinap The changes made for GPS_RESCUE silently increased gpsSol.llh.alt from 0.1m/lsb to 0.01m/lsb. So the limit would be 655m or 327m.